### ISSUE TITLE
HTTPS doesn't work when using Addressable::URI

### ISSUE THREAD

**Issue by col** [lostisland/faraday#318]
This works.
```ruby
Faraday.new(:url => 'https://www.google.com').get('/')
```

This doesn't but probably should.
```ruby
Faraday.new(:url => Addressable::URI.parse('https://www.google.com')).get('/')
```

I tracked this down to net_http.rb
```ruby
def net_http_connection(env)
  if proxy = env[:request][:proxy]
    Net::HTTP::Proxy(proxy[:uri].host, proxy[:uri].port, proxy[:user], proxy[:password])
  else
    Net::HTTP
  end.new(env[:url].host, env[:url].port)
end
```

env[:url].port returns nil when using Addressable::URI. 

I'd like to submit a pull request but I'm not actually sure what the best approach would be for supporting Addressable::URI. Thoughts?

**Comment by juanibiapina** [comment]
Was a workaround ever found? Can I use Addressable with Faraday somehow?

**Comment by iMacTia** [comment]
@juanibiapina this was fixed with #514 so now if you pass an `Addressable::URI` using https to Faraday and Net::HTTP adapter, this will use port 443.
You can see the fix [here](https://github.com/lostisland/faraday/blob/master/lib/faraday/adapter/net_http.rb#L95)
